app/assets/javascripts/thredded.es6

```diff
@@ -1,18 +1,59 @@
+//= require babel/polyfill
 //= require jquery
 //= require jquery_ujs
 //= require_tree ./thredded/vendor
 //= require_tree ./thredded
 
-class Thredded {
-  constructor() {
-    this.topicForm = new ThreddedTopicForm();
-    this.privateTopicForm = new ThreddedPrivateTopicForm();
-    this.postForm = new ThreddedPostForm();
-    this.kbShortcuts = new ThreddedKeyboardShortcuts();
-    this.timestamps = new ThreddedTimeStamps();
-    this.searchBar = new ThreddedSearchBar();
-    this.currentlyOnline = new ThreddedCurrentlyOnline();
+(function() {
+
+  const pageComponents = [
+    ['thredded-edit-post', 'postForm'],
+    ['thredded-new-private-topic', 'postForm', 'privateTopicForm'],
+    ['thredded-new-topic', 'postForm', 'topicForm'],
+    ['thredded-posts', 'postForm'],
+    ['thredded-private-topics-index', 'privateTopicForm'],
+    ['thredded-theme', 'topicForm', 'privateTopicForm', 'postForm', 'searchBar'],
+    ['thredded-topics-index', 'topicForm']
+  ];
```

Comment:
Just pushed an amended commit that fixes this, and reverts changes to schema files (not sure how that happened).